lib/monitor.js

```diff
@@ -134,57 +124,58 @@ class Monitor {
             });
         });
 
-        this._state.handlers.log = this._logHandler;
-        this._state.handlers['request-error'] = this._errorHandler;
-        this._state.handlers[this.settings.responseEvent] = this._responseHandler;
-        this._state.handlers.request = this._requestLogHandler;
+        this._state.report = true;
 
         // Initialize Events
-        internals.forOwn(this._state.handlers, (event, handler) => {
-
-            this._server.on(event, handler);
-        });
+        this._server.on('log', this._logHandler);
+        this._server.on('request-error', this._errorHandler);
+        this._server.on(this.settings.responseEvent, this._responseHandler);
+        this._server.on('request', this._requestLogHandler);
+
+        if (this.settings.wreck) {
+            const wreck = Symbol.for('wreck');
+            process[wreck].on('response', this._wreckHandler);
+        }
 
         if (this._ops) {
             this._ops.on('ops', this._opsHandler);
-            this._ops.on('error', (err) => {
-
-                console.error(err);
-            });
+            this._ops.on('error', console.error);
         }
 
         // Events can not be any of ['log', 'request-error', 'ops', 'request', 'response', 'tail']
         for (let i = 0; i < this.settings.extensions.length; ++i) {
             const event = this.settings.extensions[i];
-            const handler = this._extensionHandler.bind(this, event);
+            const self = this;
+            // Can't use () => because of "arguments"
+            this._server.on(this.settings.extensions[i], function () {
 
-            this._state.extensions[event] = handler;
-            this._server.on(this.settings.extensions[i], handler);
+                const args = Array(arguments.length);
+                for (let j = 0; j < arguments.length; ++j) {
+                    args[j] = arguments[j];
+                }
+                const payload = {
+                    event,
+                    timestamp: Date.now(),
+                    payload: args
+                };
+                self.push(() => Object.assign({}, payload));
```

Comment:
Yes. It's as close as I'm willing to do for something that gets called so often.